Zend/zend_inheritance.c

```diff
@@ -1902,11 +1909,7 @@ static void zend_add_trait_method(zend_class_entry *ce, zend_string *name, zend_
 				ZSTR_VAL(ce->name), ZSTR_VAL(name),
 				ZSTR_VAL(existing_fn->common.scope->name), ZSTR_VAL(existing_fn->common.function_name));
 		} else {
-			/* Inherited members are overridden by members inserted by traits.
-			 * Check whether the trait method fulfills the inheritance requirements. */
-			do_inheritance_check_on_method(
-				fn, fixup_trait_scope(fn, ce), existing_fn, fixup_trait_scope(existing_fn, ce),
-				ce, NULL, /* check_visibility */ 1);
+			check_inheritance = true;
```

Comment:
The movement of this call seems not necessary, because the function entry was duplicated into temporary buffer by ``zend_traits_copy_functions()``.

I'll verify if it was really necessary for ``master`` (see d344fe06a23c)

Reply:
According to my check, this wasn't necessary. So I'm not sure if it's better to keep the call in the old place or move it down.

@TimWolla ^^

Reply:
Thank you for the ping. However I cannot comment meaningfully, due to my lack of expertise on the engine. You are the expert and I trust you here: If all tests pass, then the change is probably fine.

Reply:
@TimWolla the related code was changed so many times, that I spent hours trying to understand all the combinations of conditions... :)

Two questions:

1) Do you remember why did you move call to ``do_inheritance_check_on_method_ex()`` in ``zend_add_trait_method()`` down (see commit d344fe06a23c)

2) I think ``Zend/tests/attributes/override/016.phpt`` is wrong. It should emit fatal error, because class ``C`` don't have parent method at all.

Reply:
> 1. Do you remember why did you move call to `do_inheritance_check_on_method_ex()` in `zend_add_trait_method()` down (see commit [d344fe0](https://github.com/php/php-src/commit/d344fe06a23c423a183b0b328da27005274768ed))

I don't remember exactly. Looking at the commit history of the PR, I made the change in:

f5fd079ef811fb37bf9a8767cff4573e29f7da99

The direct parent of it (66a71119380ea45b9a9bc7182b4ed4972de5b710) failed the ASAN CI. Unfortunately the logs are no longer available.

However based on that information, I assume that my initial attempt at the fix caused some memory unsafety that I was able to fix by moving the call down. Perhaps a future follow-up (e.g. Ilija's review suggestion) fixed it properly such that the moving of the call is no longer required, I cannot comment on that.

> 2\. I think `Zend/tests/attributes/override/016.phpt` is wrong. It should emit fatal error, because class `C` don't have parent method at all.

It does not have a parent method, but its signature is enforced by the `abstract` method in the trait. This has been voted on as part of the RFC.

Reply:
Thank you!